apps/condo/domains/billing/access/BillingReceipt.js

```diff
@@ -2,10 +2,34 @@
  * Generated by `createschema billing.BillingReceipt 'context:Relationship:BillingIntegrationOrganizationContext:CASCADE; importId?:Text; property:Relationship:BillingProperty:CASCADE; account:Relationship:BillingAccount:CASCADE; period:CalendarDay; raw:Json; toPay:Text; services:Json; meta:Json'`
  */
 
-const { canManageBillingEntityWithContext, canReadBillingEntity } = require('@condo/domains/billing/utils/accessSchema')
+const { Resident, ServiceConsumer } = require('@condo/domains/resident/utils/serverSchema')
+const { canManageBillingEntityWithContext } = require('@condo/domains/billing/utils/accessSchema')
 
-async function canReadBillingReceipts ({ authentication: { item: user } }) {
-    return await canReadBillingEntity(user)
+async function canReadBillingReceipts ({ context, authentication: { item: user } }) {
+
+    async function _getBillingAccountByUser (user) {
+        const [ resident ] = await Resident.getAll(context, { user: { id: user.id } })
+        if (!resident) return null
+
+        const [ serviceConsumer ] = await ServiceConsumer.getAll(context, { resident: { id: resident.id } })
+        if ( !serviceConsumer ) return null
+
+        if (!serviceConsumer.billingAccount) return null
+        return serviceConsumer.billingAccount.id
+    }
+
+    if (!user) return false
+    if (user.isAdmin) return true
+
+    let billingAccountId = await _getBillingAccountByUser(user)
+
+    return {
+        OR: [
+            { context: { organization: { employees_some: { user: { id: user.id }, role: { canManageIntegrations: true }, deletedAt: null, isBlocked: false } } } },
+            { context: { integration: { accessRights_some: { user: { id: user.id } } } } },
+            { account: { id: billingAccountId } },
```

Comment:
i think it's better to move it to custom query rather than to give access to residents to raw billing data